core/arch/arm/mm/core_mmu.c

```diff
@@ -379,6 +379,10 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 		carve_out_phys_mem(&m, &num_elems, pmem->addr, pmem->size);
 #endif
 
+#ifdef CFG_DT
+	carve_out_phys_mem(&m, &num_elems, CFG_TZDRAM_START, CFG_TZDRAM_SIZE);
```

Comment:
This function is within `#ifdef CFG_CORE_DYN_SHM`, we need to add `CFG_DT` to that.